share/wake/lib/system/path.wake

```diff
@@ -18,8 +18,7 @@ package wake
 # Read the file contents of a Path
 export def read (path: Path): Result String Error =
     def imp p = prim "read"
-    require Pass f = getPathResult path
-    match (imp f.getPathName)
+    match (imp path.getPathName)
```

Comment:
Again, I think this one needs to be defensive until BadPath is really gone.

Reply:
It might even be better to keep it defensive even after removing `BadPath` -- a good `Path` only proves that the `Path` existed when it was created, but things could theoretically be deleted between creation and use if the user/system is doing bad things.  I don't know how much of a performance hit it would be (and almost *any* hit would probably be too much), but once `getPathResult` isn't needed as a simple constructor test, would we want to make it a "is this file *still* good" check?

Reply:
There is nothing the build can do to protect itself from users who modify the workspace while a build is running. This is why I don't want to use developer builds as a shared cache publisher. ;-) As long as no person is manipulating the workspace during the build, though, you should be fine; wake-fuse protects all extant files during Job execution.

Reply:
So, once BadPath is removed from the `Path` type, I would indeed remove this along the lines of your original PR.